packages/x-data-grid/src/components/grid/toolbar/GridToolbarToggleButton.tsx

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable react/no-unused-prop-types */
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import ToggleButton, { ToggleButtonProps } from '@mui/material/ToggleButton';
+import { styled } from '@mui/system';
+import composeClasses from '@mui/utils/composeClasses';
+import clsx from 'clsx';
+import { useGridRootProps } from '../../../hooks/utils/useGridRootProps';
+import { getDataGridUtilityClass } from '../../../constants/gridClasses';
+import type { DataGridProcessedProps } from '../../../models/props/DataGridProps';
+import {
+  useGridComponentRenderer,
+  RenderProp,
+} from '../../../hooks/utils/useGridComponentRenderer';
+
+export interface GridToolbarToggleButtonProps extends ToggleButtonProps {
```

Comment:
I don't love this pattern, I already need to remove it from the rest of the codebase for the design-system agnostic refactor. Is your plan to expose material-ui props directly on these components?

My plan at the moment is to write a minimal/simplified list of props we use in the grid, and expose only those on the base slots typings.

Reply:
Shouldn't the `render={<Something />}` pattern be enough to let users use material-ui props on these?

Reply:
From our discussion yesterday, we'll have an unstyled version of this component that doesn't import anything from Material UI, and then a wrapper component for Material UI that uses the unstyled version.

Given that eventually, we'll have dozens of these components, the need for a wrapper for each one doesn't sound great...

Reply:
After our discussion last week, I made some changes to the way the `Grid.Toolbar` components get rendered. We discussed having an unstyled version of each component, and a material version, but after exploring this route it felt like a lot of duplicated code.

An option I have been experimenting with since then is having each of the `Grid.Toolbar` components render a base slot e.g. https://github.com/mui/mui-x/pull/14611/files#diff-9927b5c02d2b489616da79753dd06b16dcd1db100363227f31ce632fb7efeae6. The base slots defined for material: https://github.com/mui/mui-x/pull/14611/files#diff-ddff2463931b160b1a96800f1c8eccd98ae2f326e550ad7a60601868dcc08b3a.

With this approach, we have a clear separation of concerns:

- **Base slots** are styled specific to the design system and can be used in composable parts.
- **Composable parts** are design agnostic components that provide context, event handlers and other attributes to the rendered element. The default rendered element is a base slot e.g. `baseButton`, which can be overwritten in user land via the `slots` prop. The rendered element can also be overwritten on an individual basis via the `render` prop.